subprojects/core-api/src/main/java/org/gradle/process/InputOutputValue.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.tasks.Internal;
+
+/**
+ * An input or output value.
+ *
+ * Subtypes should be annotated with the corresponding input/output annotations.
+ *
+ * @since 4.7
+ */
+@Incubating
+public interface InputOutputValue {
+    @Internal
+    Object getValue();
```

Comment:
Why is declaring `getValue()` useful on the public API? If we need the method, we should probably do it on `InputOutputValueInternal`.

Reply:
I changed the interface to be `HasStringRepresentation`. I think it should be fine if plugins implement `HasStringRepresentation`.

The `getValue()` method moved to the internal types.